packages/internal-test-utils/__tests__/ReactInternalTestUtils-test.js

```diff
@@ -301,3 +307,2079 @@ describe('ReactInternalTestUtils console mocks', () => {
+      });
+      expect(message).toMatchInlineSnapshot(`
+        "assertConsoleLogDev(expected)
+
+        Expected log was not recorded.
+
+        - Expected logs
+        + Received logs
+
+          Hi
+          Wow
+        - Bye"
+      `);
+    });
+
+    // @gate __DEV__
+    it('fails if first received log is not included', () => {
+      const message = expectToWarnAndToThrow(() => {
+        console.log('Hi');
+        console.log('Wow');
+        console.log('Bye');
+        assertConsoleLogDev(['Wow', 'Bye']);
+      });
+      expect(message).toMatchInlineSnapshot(`
+        "assertConsoleLogDev(expected)
+
+        Unexpected log(s) recorded.
+
+        - Expected logs
+        + Received logs
+
+        + Hi
+          Wow
+          Bye"
+      `);
+    });
+
+    // @gate __DEV__
+    it('fails if middle received log is not included', () => {
+      const message = expectToWarnAndToThrow(() => {
+        console.log('Hi');
+        console.log('Wow');
+        console.log('Bye');
+        assertConsoleLogDev(['Hi', 'Bye']);
+      });
+      expect(message).toMatchInlineSnapshot(`
+        "assertConsoleLogDev(expected)
+
+        Unexpected log(s) recorded.
+
+        - Expected logs
+        + Received logs
+
+          Hi
+        + Wow
+          Bye"
+      `);
+    });
+
+    // @gate __DEV__
+    it('fails if last received log is not included', () => {
+      const message = expectToWarnAndToThrow(() => {
+        console.log('Hi');
+        console.log('Wow');
+        console.log('Bye');
+        assertConsoleLogDev(['Hi', 'Wow']);
+      });
+      expect(message).toMatchInlineSnapshot(`
+        "assertConsoleLogDev(expected)
+
+        Unexpected log(s) recorded.
+
+        - Expected logs
+        + Received logs
+
+          Hi
+          Wow
+        + Bye"
+      `);
+    });
+
+    // @gate __DEV__
+    it('fails if both expected and received mismatch', () => {
+      const message = expectToWarnAndToThrow(() => {
+        console.log('Hi');
+        console.log('Wow');
+        console.log('Bye');
+        assertConsoleLogDev(['Hi', 'Wow', 'Yikes']);
+      });
+      expect(message).toMatchInlineSnapshot(`
+        "assertConsoleLogDev(expected)
+
+        Unexpected log(s) recorded.
+
+        - Expected logs
+        + Received logs
+
+          Hi
+          Wow
+        - Yikes
+        + Bye"
+      `);
+    });
+
+    // @gate __DEV__
+    it('fails if both expected and received mismatch with multiple lines', () => {
+      const message = expectToWarnAndToThrow(() => {
+        console.log('Hi\nFoo');
+        console.log('Wow\nBar');
+        console.log('Bye\nBaz');
+        assertConsoleLogDev(['Hi\nFoo', 'Wow\nBar', 'Yikes\nFaz']);
+      });
+      expect(message).toMatchInlineSnapshot(`
+        "assertConsoleLogDev(expected)
+
+        Unexpected log(s) recorded.
+
+        - Expected logs
+        + Received logs
+
+          Hi Foo
+          Wow Bar
+        - Yikes Faz
```

Comment:
Shouldn't this include the newlines?

Reply:
No, I strip the new lines to make the diffs better, otherwise they could get really fucky and useless